### PR TITLE
LPS-137866 Validate integer and long type size to prevent wrong data save

### PR DESCRIPTION
**Pull request by mateussgn**
Related issue: https://issues.liferay.com/browse/LPS-137866

### CODE REVIEW

**Comment by mateussgn** [comment]
ci:test:sf

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/display/view.jsp [hunk 1]
```diff
@@ -179,7 +179,10 @@ long formInstanceId = ddmFormDisplayContext.getFormInstanceId();
 						<liferay-ui:error exception="<%= NoSuchFormInstanceException.class %>" message="the-selected-form-no-longer-exists" />
 						<liferay-ui:error exception="<%= NoSuchStructureException.class %>" message="unable-to-retrieve-the-definition-of-the-selected-form" />
 						<liferay-ui:error exception="<%= NoSuchStructureLayoutException.class %>" message="unable-to-retrieve-the-layout-of-the-selected-form" />
-						<liferay-ui:error exception="<%= ObjectEntryValuesException.class %>" message="the-maximum-length-is-280-characters-for-text-fields" />
+						<liferay-ui:error exception="<%= ObjectEntryValuesException.IntegerSizeExceeded.class %>" message="object-entry-value-exceeds-integer-field-size" />
+						<liferay-ui:error exception="<%= ObjectEntryValuesException.LongMaxSizeExceeded.class %>" message="Object entry value exceeds maximum Long field allowed size." />
```
Comment (marcelabc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/display/view.jsp:183]: Would you change the message for the language key, please?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/display/view.jsp [hunk 2]
```diff
@@ -179,7 +179,10 @@ long formInstanceId = ddmFormDisplayContext.getFormInstanceId();
...
+						<liferay-ui:error exception="<%= ObjectEntryValuesException.LongMinSizeExceeded.class %>" message="Object entry value exceeds minimum Long field allowed size." />
```
Comment (marcelabc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/display/view.jsp:184]: Would you change the message for the language key, please?

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 3]
```diff
@@ -10145,6 +10145,9 @@ object-action-trigger[onAfterUpdate]=On After Update
 object-definition-data=Object Data
 object-definition-id=Object ID
 object-definition-table-name=Table Name
+object-entry-value-exceeds-integer-field-size=Object entry value exceeds Integer field allowed size.
```
Comment (marcelabc) [inline modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:10148]: Could you fix the language key with all the words contained in the message?

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java [hunk 4]
```diff
@@ -1558,16 +1559,56 @@ private void _validateListTypeEntriesValues(
...
+
+		String entryValueString = entryValue.toString();
+
+		if (!entryValueString.isEmpty()) {
+			long value = GetterUtil.getLong(entryValue);
+
+			if (!StringUtil.equals(
+					String.valueOf(value), entryValue.toString())) {
+
+				throw new ObjectEntryValuesException.LongMaxSizeExceeded();
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:1593]: Could you create a generic message for this case, please?

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hey @mateussgn, thanks for the hard work! There are some minor changes, could you review them, please? Let me know if you have any questions, thanks 😄

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/display/view.jsp (hunk 1 above)
Comment (mateussgn) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/display/view.jsp:183]: Sure!
Thanks for the tip!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/display/view.jsp (hunk 2 above)
Comment (mateussgn) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/display/view.jsp:184]: Sure!
Thanks for the tip!

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties (hunk 3 above)
Comment (mateussgn) [reply modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:10148]: Sure!
Thanks for the tip!

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java (hunk 4 above)
Comment (mateussgn) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:1593]: Sure!
Thanks for the tip!

**Comment by mateussgn** [comment]
ci:test:sf

**Comment by mateussgn** [comment]
ci:test:sf

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:stop

**Comment by marcelabc** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/display/view.jsp [hunk 5]
```diff
@@ -179,7 +179,11 @@ long formInstanceId = ddmFormDisplayContext.getFormInstanceId();
...
 						<liferay-ui:error exception="<%= NoSuchStructureException.class %>" message="unable-to-retrieve-the-definition-of-the-selected-form" />
 						<liferay-ui:error exception="<%= NoSuchStructureLayoutException.class %>" message="unable-to-retrieve-the-layout-of-the-selected-form" />
-						<liferay-ui:error exception="<%= ObjectEntryValuesException.class %>" message="the-maximum-length-is-280-characters-for-text-fields" />
+						<liferay-ui:error exception="<%= ObjectEntryValuesException.IntegerSizeExceeded.class %>" message="object-entry-value-exceeds-integer-field-allowed-size" />
+						<liferay-ui:error exception="<%= ObjectEntryValuesException.LongMaxSizeExceeded.class %>" message="object-entry-value-exceeds-maximum-long-field-allowed-size" />
+						<liferay-ui:error exception="<%= ObjectEntryValuesException.LongMinSizeExceeded.class %>" message="object-entry-value-exceeds-minimum-long-field-allowed-size" />
+						<liferay-ui:error exception="<%= ObjectEntryValuesException.LongSizeExceeded.class %>" message="object-entry-value-exceeds-integer-field-allowed-size" />
```
Comment (marcelabc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/display/view.jsp:185]: Can you change the message to match the exception, please?

**Review by marcelabc — CHANGES_REQUESTED** [review]

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/display/view.jsp (hunk 5 above)
Comment (mateussgn) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/resources/META-INF/resources/display/view.jsp:185]: Thanks for noticing it!! Fixing right now!

**Comment by mateussgn** [comment]
ci:stop

**Comment by mateussgn** [comment]
ci:test:sf

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
ci:test:relevant

**Review by marcelabc — APPROVED** [review]
LGTM 🎉

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
After this last test analysis, I'll rebase the branch and open a new pull request.

**Comment by mateussgn** [comment]
ci:close

**Comment by mateussgn** [comment]
Resent on https://github.com/liferay-objects/liferay-portal/pull/317

**Pull request closed without merging**